src/sdl/sdl_glimp.c

```diff
@@ -973,10 +974,17 @@ void GLimp_Splash(glconfig_t *glConfig)
 	dstRect.h = splashImage->h;
 
 	// apply image on surface
-	SDL_BlitSurface(splashImage, NULL, SDL_GetWindowSurface(main_window), &dstRect);
-	SDL_UpdateWindowSurface(main_window);
+	blitResult = SDL_BlitSurface(splashImage, NULL, SDL_GetWindowSurface(main_window), &dstRect);
 
-	SDL_FreeSurface(splashImage);
+	if (blitResult < 0)
+	{		
+		Com_Printf(S_COLOR_YELLOW "SDL_BlitSurface failed - %s\n", SDL_GetError());
+		return;			
```

Comment:
I suppose you need `SDL_FreeSurface(splashImage)` before the return here, but otherwise LGTM!

Reply:
Actually, might be even better to write something like:
```
if (blitResult >= 0)
{
    SDL_UpdateWindowSurface(main_window);
}
```

so `SDL_FreeSurface` will be called thereafter.